addons/udes_common/models/models.py

```diff
@@ -134,3 +135,43 @@ def selection_display_name(self, selection_field_name):
     # Odoo has a nice function that handles all the weird ways you can define selections!
     selection_values_dict = dict(selection_field._description_selection(self.env))
     return selection_values_dict.get(self[selection_field_name], False)
+
+
+class Base(models.AbstractModel):
+    _inherit = 'base'
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        res = super().fields_view_get(view_id, view_type, toolbar, submenu)
+        doc = etree.XML(res["arch"])
+        if self._context.get('view_all_fields'):
+            # Hide the view more button
+            doc = etree.XML(res["arch"])
+            for node in doc.xpath("//button[@name='view_more_info']"):
+                node.getparent().remove(node)
+        else:
+            hidden_fields_from_form_view = tools.hidden_fields_from_form_view.get(self._name, False)
+            if not hidden_fields_from_form_view:
+                return res
```

Comment:
Nit: I always wondered if a function should have one exit point or not, I cannot argue which is better except that having one exit point seems a little cleaner. Are you returning res here to avoid having multiple if statements?

I would suggest simplifying it as follows:

if hidden_fields_from_form_view and view_type == "form":
    for field_name in hidden_fields_from_form_view:
    ...

Reply:
Sometimes where we need to get out without running extra lines we may have more than one returns in the methods.
I think still is the case because at the end is this line which i didnt want to repeat more than once 
`res["arch"] = etree.tostring(doc)` i thought is better to return res in case there wont be any changes

Reply:
I think that i responded in a logic way the reason why i am returning , just not to execute any further code which is after and unnecessary to run. 
You are saying that you think should be only one return which i agree at some point but we are using in many cases in our custom modules and in core modules more than one return in a method, soi hope you should be happy to resolve this thread without any change.

Reply:
Sounds good, I wanted to spark a discussion more than forcing my nit 😁